src/fabric_cli/utils/fab_cmd_mkdir_utils.py

```diff
@@ -405,12 +413,47 @@ def _validate_credential_params(cred_type, provided_cred_params):
         utils_ui.print_warning(
             f"Ignoring unsupported parameters for credential type {cred_type}: {ignored_params}"
         )
+    if is_on_premises_gateway:
+        provided_cred_params["values"] = _validate_and_get_on_premises_gateway_credential_values(provided_cred_params.get("values"))
 
     for key in param_keys:
         params[key] = provided_cred_params[key.lower()]
 
     return params
 
+def _validate_and_get_on_premises_gateway_credential_values(cred_values):
+    for item in cred_values:
+        if not isinstance(item, dict):
+            raise FabricCLIError(
+                ErrorMessages.Common.invalid_onpremises_gateway_values(),
+                fab_constant.ERROR_INVALID_INPUT,
+            )
+        
+    param_values_keys = ["gatewayId", "encryptedCredentials"]
+    missing_params = [
+        key for key in param_values_keys 
+        if not all(key.lower() in {k.lower() for k in item.keys()} for item in cred_values)
+    ]
+    if len(missing_params) > 0:
+        raise FabricCLIError(
+            ErrorMessages.Common.missing_onpremises_gateway_parameters(missing_params),
+            fab_constant.ERROR_INVALID_INPUT,
+    )
+
+    ignored_params = [
+        key
+        for item in cred_values
+        for key in item.keys()
+        if key not in [k.lower() for k in param_values_keys]
+    ]
+    if len(ignored_params) > 0:
+        utils_ui.print_warning(
```

Comment:
why do we print it as warning? worth considering as debug. can you explain the reason?